src/modules/deal-investor/deal-investor.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@nestjs/common';
+import { DateStringTransformations, UkefId } from '@ukef/helpers';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs/acbs-authentication.service';
+import { AcbsDealPartyService } from '@ukef/modules/acbs/acbs-deal-party.service';
+import { AcbsResourceNotFoundException } from '@ukef/modules/acbs/exception/acbs-resource-not-found.exception';
+
+import { DealInvestor } from './deal-investor.interface';
+
+@Injectable()
+export class DealInvestorService {
+  // TODO: make DateStringTransformations injectable, maybe it needs to be part of module for this to work.
+  private readonly dateStringTransformations: DateStringTransformations = new DateStringTransformations();
+
+  constructor(private readonly acbsAuthenticationService: AcbsAuthenticationService, private readonly acbsDealPartyService: AcbsDealPartyService) {}
+
+  async getDealInvestors(dealIdentifier: UkefId): Promise<DealInvestor[]> {
```

Comment:
Is `DealInvestor[]` not the same as `DealInvestorResponseDto`?

Reply:
Other module had interface, so I did the same thing here. DealInvestor interface is similar to DealInvestorResponseDto class.
We could merge or remove one of them later?
I am not sure if it is required, maybe it better to define type using interface?
@lmarrai-sw

Reply:
If they are same and share similar properties then best we avoid repeating code.

Reply:
I think it's good practice to separate the request/response types from the types that you use to pass data around in the code itself because these things will change for different reasons.

E.g. we might implement another endpoint that needs to get extra fields for deal investors which we'd modify this service method for, and then we can change the `DealInvestor` type without affecting `DealInvestorResponseDto` (so we don't affect the response from this endpoint)

If you think that's overkill to do upfront then we could flatten it down to 1 interface for now and separate them if needed later?

Reply:
Currently each endpoint has unique types, not sure if we need to define separate internal and external types for each endpoint.

Also maybe GET and POST maybe should share same object structure to simplify application.

Reply:
Personally I'm in the 'separate them if needed later' camp, I think having separate types adds to complexity and is probably YAGNI

Reply:
Lets merge this PR without change. 
It can be reviewed when working on POST for same deal/investor, maybe interface will be useful.
Plus this is global question how all code should be written.